src/api/java/baritone/api/utils/ExampleBaritoneControl.java

```diff
@@ -672,8 +675,9 @@ public boolean isInGoal(int x, int y, int z) {
         return false;
     }
 
-    private int parseOrDefault(String str, int i) {
-        return str.equals("~") ? i : str.startsWith("~") ? Integer.parseInt(str.substring(1)) + i : Integer.parseInt(str);
+    // dimensionFactor is so the coords are converted
+    private int parseOrDefault(String str, int i, double dimensionFactor) {
+        return str.equals("~") ? i : str.startsWith("~") ? Integer.parseInt(str.substring(1)) + i : (int) (Integer.parseInt(str) * dimensionFactor);
```

Comment:
why doesn't the dimension factor apply? if i'm currently in the nether and i do `goal ~100 ~ ~100 overworld` shouldn't it add 100 to my overworld coordinates? i feel like `i` could be `(int) (i * dimensionFactor)`

Reply:
well, when you just do ~ I think you wouldnt want to go somewhere further out then your coordinate. i is the coordinate where you are right now so I think that should stay. If you add 100 now to your overworld coordinates it would be nether coords + 100 / 8 to go there. So the string that is getting parsed should be multiplied in both times. Does that make sense?